waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -520,6 +547,22 @@ proc pruneInRelayConns(pm: PeerManager, amount: int) {.async.} =
   for p in inRelayPeers[0..<connsToPrune]:
     await pm.switch.disconnect(p)
 
+proc pruneConnsByIp*(pm: PeerManager) {.async.} =
+  ## prunes connections based on ip colocation, allowing no more
+  ## than ColocationLimit inbound connections from same ip
+  ##
+
+  # update the table tracking ip and the connected peers
+  pm.updateIpTable()
```

Comment:
Mmm. This rebuilds the IP table every time, so I don't see any reason to cache the `ipTable` on the `pm` object. In other words, why not have a `getPeersPerIp` that simply creates and populates such a table every time?

Reply:
yup, but in the future I plan to use this table for other things. still unclear but some ideas:
* intercept a connection. before even allowing a connection to go through, check if colocation limit is ok.
* discovery. if we discover a peer violating the colocation limit, do not add it to the peer store.
* out connections. dont start connections to peers violating the colocation limit.

so would like to leave it if its fine.